packages/migrate/src/commands/DbExecute.ts

```diff
@@ -0,0 +1,171 @@
+    const args = arg(
+      argv,
+      {
+        '--help': Boolean,
+        '-h': '--help',
+        '--stdin': Boolean,
+        '--file': String,
+        '--schema': String,
+        '--url': String,
+        '--preview-feature': Boolean,
+        '--telemetry-information': String,
+      },
+      false,
+    )
+
+    if (isError(args)) {
+      return this.help(args.message)
+    }
+
+    if (args['--help']) {
+      return this.help()
+    }
+
+    if (!args['--preview-feature']) {
+      throw new DbExecuteNeedsPreviewFeatureFlagError()
+    }
+
+    // One of --stdin or --file is required
+    if (args['--stdin'] && args['--file']) {
+      throw new Error(
+        `--stdin and --file cannot be used at the same time. Only 1 must be provided. 
+See \`${chalk.green(getCommandWithExecutor('prisma db execute -h'))}\``,
+      )
+    } else if (!args['--stdin'] && !args['--file']) {
+      throw new Error(
+        `Either --stdin or --file must be provided.
+See \`${chalk.green(getCommandWithExecutor('prisma db execute -h'))}\``,
+      )
+    }
+
+    // One of --url or --schema is required
+    if (args['--url'] && args['--schema']) {
+      throw new Error(
+        `--url and --schema cannot be used at the same time. Only 1 must be provided.
+See \`${chalk.green(getCommandWithExecutor('prisma db execute -h'))}\``,
+      )
+    } else if (!args['--url'] && !args['--schema']) {
+      throw new Error(
+        `Either --url or --schema must be provided.
+See \`${chalk.green(getCommandWithExecutor('prisma db execute -h'))}\``,
+      )
+    }
+
+    let script = ''
+    // Read file
+    if (args['--file']) {
+      script = fs.readFileSync(path.resolve(args['--file']), 'utf-8')
+    }
+    // Read stdin
+    if (args['--stdin']) {
+      script = await getStdin()
+      // If input is empty, stop here
+      if (!script) {
+        throw new Error(
+          `--stdin was passed but the input was empty. See \`${chalk.green(
+            getCommandWithExecutor('prisma db execute -h'),
+          )}\``,
+        )
+      }
+    }
+
+    let datasourceType: EngineArgs.DbExecuteDatasourceType
+
+    const schemaPath = await getSchemaPath(args['--schema'])
+    if (!schemaPath) {
+      throw new NoSchemaFoundError()
+    }
+
+    // Execute command(s) to url passed
+    if (args['--url']) {
+      datasourceType = {
+        tag: 'url',
+        url: args['--url'],
+      }
+    } else {
+      // Execute command(s) to url from schema
+      datasourceType = {
+        tag: 'schema',
+        schema: schemaPath,
+      }
+    }
+
+    // console.info(chalk.dim(`Prisma schema loaded from ${path.relative(process.cwd(), schemaPath)}`))
+    // await printDatasource(schemaPath)
+    const migrate = new Migrate()
+
+    try {
+      await migrate.engine.dbExecute({
+        script,
+        datasourceType,
+      })
+    } finally {
+      migrate.stop()
+    }
+
+    return `Script executed successfully.`
```

Comment:
I don't see where the value is used — it doesn't look like it's printed.

Reply:
It's printed from here https://github.com/prisma/prisma/blob/integration/cli-db-execute/packages/cli/src/bin.ts#L177:L177